src/parser/grammar.lalrpop

```diff
@@ -863,15 +863,15 @@ NOpPre<ArgRule>: UniTerm = {
 }
 
 TypeBuiltin: Types = {
-    "Dyn" => Types(TypeF::Dyn),
-    "Num" => Types(TypeF::Num),
-    "Bool" => Types(TypeF::Bool),
-    "Str" => Types(TypeF::Str),
+    <l: @L> "Dyn" <r: @R> => Types { types: TypeF::Dyn, pos: mk_pos(src_id, l, r) },
+    <l: @L> "Num" <r: @R>  => Types { types: TypeF::Num, pos: mk_pos(src_id, l, r) },
+    <l: @L> "Bool" <r: @R>  => Types { types: TypeF::Bool, pos: mk_pos(src_id, l, r) },
+    <l: @L> "Str" <r: @R>  => Types { types: TypeF::Str, pos: mk_pos(src_id, l, r) },
 }
 
 TypeAtom: Types = {
     <TypeBuiltin>,
```

Comment:
In fact if `TypeBuiltin` is only used in `TypeAtom`, you don't even need a `WithPos`, as the one for `TypeAtom` will take care of setting the position.

But you still need to look for any usage of `TypeAtom`, and replace that with `WithPos<TypeAtom>`.